core/shared/src/main/scala/scalaz/zio/ZManaged.scala

```diff
@@ -73,93 +211,573 @@ final case class ZManaged[-R, +E, +A](reserve: ZIO[R, E, Reservation[R, E, A]])
+                  .flatMap(res => restore(res.acquire))
+              }
+            val onFailure = (e: E) =>
+              ZIO.uninterruptibleMask { restore =>
+                failure(e).reserve
+                  .flatMap(res => finalizers.update(fs => res.release *> fs).const(res))
+                  .flatMap(res => restore(res.acquire))
+              }
+            val onSuccess = (a: A) =>
+              ZIO.uninterruptibleMask { restore =>
+                success(a).reserve
+                  .flatMap(res => finalizers.update(fs => res.release *> fs).const(res))
+                  .flatMap(res => restore(res.acquire))
+              }
+            direct.foldM(onFailure, onSuccess)
           },
           release = ZIO.flatten(finalizers.get)
         )
       }
     }
 
-  final def <&>[R1 <: R, E1 >: E, A1](that: ZManaged[R1, E1, A1]): ZManaged[R1, E1, (A, A1)] =
-    zipWithPar(that)((_, _))
+  /**
+   * Unwraps the optional success of this effect, but can fail with unit value.
+   */
+  final def get[E1 >: E, B](implicit ev1: E1 =:= Nothing, ev2: A <:< Option[B]): ZManaged[R, Unit, B] =
+    ZManaged.absolve(mapError(ev1).map(ev2(_).toRight(())))
+
+  /**
+   * Depending on the environment execute this or the other effect
+   */
+  final def join[R1, E1 >: E, A1 >: A](that: ZManaged[R1, E1, A1]): ZManaged[Either[R, R1], E1, A1] = self ||| that
+
+  final def left[R1 <: R, C]: ZManaged[Either[R1, C], E, Either[A, C]] = self +++ ZManaged.identity
+
+  /**
+   * Returns an effect whose success is mapped by the specified `f` function.
+   */
+  final def map[B](f0: A => B): ZManaged[R, E, B] =
+    ZManaged[R, E, B] {
+      reserve.map(token => token.copy(acquire = token.acquire.map(f0)))
+    }
+
+  /**
+   * Returns an effect whose failure is mapped by the specified `f` function.
+   */
+  final def mapError[E1](f: E => E1): ZManaged[R, E1, A] =
+    Managed(reserve.mapError(f).map(r => Reservation(r.acquire.mapError(f), r.release)))
+
+  /**
+   * Executes this effect, skipping the error but returning optionally the success.
+   */
+  final def option: ZManaged[R, Nothing, Option[A]] =
+    fold(_ => None, Some(_))
+
+  /**
+   * Translates effect failure into death of the fiber, making all failures unchecked and
+   * not a part of the type of the effect.
+   */
+  final def orDie(implicit ev: E <:< Throwable): ZManaged[R, Nothing, A] =
+    orDieWith(ev)
+
+  /**
+   * Keeps none of the errors, and terminates the fiber with them, using
+   * the specified function to convert the `E` into a `Throwable`.
+   */
+  final def orDieWith(f: E => Throwable): ZManaged[R, Nothing, A] =
+    mapError(f).catchAll(ZManaged.die)
+
+  /**
+   * Executes this effect and returns its value, if it succeeds, but
+   * otherwise executes the specified effect.
+   */
+  final def orElse[R1 <: R, E2, A1 >: A](that: => ZManaged[R1, E2, A1]): ZManaged[R1, E2, A1] =
```

Comment:
There's a `<>` alias for that combinator, I think.

Reply:
Yep. It's already in at lines 95-96. They are currently not next to each other as i've sorted everything alpabetically, which might not be the best idea in such cases?

Reply:
Oh, my bad, didn't see it 🙈